File: src/php/Generic/HashBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Generic {

    use EntityFX\NetBenchmark\Core\Writer;
    
    class HashBenchmark extends HashBase {

        protected function BenchImplementation() {
            $result = "";

            for ($i = 0; $i < $this->Iterrations; $i++)
            {
                $result = self::DoHash($i);
            }
            return $result;
        }
    }
}

File: src/lua/generic/hashBenchmark.lua
HashBenchmark = class(HashBase, function(a, writer, printToConsole)
    HashBase.init(a, writer, printToConsole)
    a.name = "HashBenchmark"
end)

function HashBenchmark:benchImplementation()
    local result = {}
    for i=1,self.iterrations do
        result = self:doHash(i, self.strs)
    end
    return result
end